lib/options.go

```diff
@@ -44,6 +45,79 @@ const DefaultScenarioName = "default"
 // nolint: gochecknoglobals
 var DefaultSummaryTrendStats = []string{"avg", "min", "med", "max", "p(90)", "p(95)"}
 
+type DNSConfig struct {
+	TTL      null.String     `json:"ttl"`
+	Strategy NullDNSStrategy `json:"strategy"`
+}
+
+//go:generate enumer -type=DNSStrategy -transform=kebab -trimprefix DNS -output dns_strategy_gen.go
+type DNSStrategy uint8
+
+type NullDNSStrategy struct {
+	DNSStrategy
+	Valid bool
+}
+
+const (
+	DNSFirst DNSStrategy = iota + 1
+	DNSRoundRobin
+	DNSRandom
+)
+
+// nolint: gochecknoglobals
+var DefaultDNSConfig = DNSConfig{
+	TTL:      null.StringFrom("inf"),
+	Strategy: NullDNSStrategy{DNSFirst, true},
+}
+
+func (c *DNSConfig) UnmarshalJSON(data []byte) error {
+	params := make(map[string]interface{})
+	if err := json.Unmarshal(data, &params); err != nil {
+		return err
+	}
+	return c.unmarshalDNSConfig(params)
+}
+
+func (c *DNSConfig) UnmarshalText(text []byte) error {
+	params, err := strvals.Parse(string(text))
+	if err != nil {
+		return err
+	}
+	return c.unmarshalDNSConfig(params)
+}
+
+func (c *DNSConfig) unmarshalDNSConfig(params map[string]interface{}) error {
+	if ttl, ok := params["ttl"]; ok {
+		c.TTL = null.StringFrom(fmt.Sprintf("%v", ttl))
+	}
+	c.Strategy = DefaultDNSConfig.Strategy
+	if strat, ok := params["strategy"]; ok {
+		if s, err := DNSStrategyString(strat.(string)); err != nil {
+			return err
+		} else {
+			c.Strategy.DNSStrategy = s
+			c.Strategy.Valid = true
+		}
+	}
+	return nil
+}
+
+func (s NullDNSStrategy) MarshalJSON() ([]byte, error) {
+	return json.Marshal(s.String())
+}
+
+func (s *NullDNSStrategy) UnmarshalJSON(data []byte) error {
+	var str string
+	if err := json.Unmarshal(data, &str); err != nil {
+		return err
+	}
+	var err error
+	if s.DNSStrategy, err = DNSStrategyString(str); err == nil {
+		s.Valid = true
+	}
+	return err
```

Comment:
it is usually better to return `nil` directly for [when it will go through a variable just in case it returns an error](https://yourbasic.org/golang/gotcha-why-nil-error-not-equal-nil/) that isn't just the interface `error`

Also, I find the code ... strange, which I noticed it

p.s. Isn't the unmarshalling of this just `string(data)` ? or does it include the quotes around it? I guess it includes the quote ... probably better to just use `json.Unsmarshal` then :thinking:

Reply:
Thanks about the `err` tip. I removed this unmarshal method in the latest changes, and strategy is now unmarshalled in the `DNSConfig` one.